From 8de8ce551cbdb1250b696a5eedc5cff2b68e71e8 Mon Sep 17 00:00:00 2001
From: Chris Leech <cleech@redhat.com>
Date: Mon, 24 Feb 2014 03:14:17 -0500
Subject: [PATCH] iscsiadm: Fix the hostno check for stats submode of host mode

This fixes the condition where hostno value of 0 was being treated
as invalid.

Signed-off-by: Chris Leech <cleech@redhat.com>
Signed-off-by: Adheer Chandravanshi <adheer.chandravanshi@qlogic.com>
---
 usr/iscsiadm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr/iscsiadm.c b/usr/iscsiadm.c
index 3cde8d1..36617ab 100644
--- a/usr/iscsiadm.c
+++ b/usr/iscsiadm.c
@@ -3470,7 +3470,7 @@ main(int argc, char **argv)
 						       &params);
 				break;
 			case MODE_HOST_STATS:
-				if (!host_no) {
+				if (host_no > MAX_HOST_NO) {
 					log_error("STATS mode requires host no");
 					rc = ISCSI_ERR_INVAL;
 					break;
-- 
1.8.3.1

